app/models/ability.rb

```diff
@@ -40,14 +41,29 @@ def assign_staff_read_permission(current_user)
     can :read, EdiInvoice if /Y/.match?(current_user.edi_inv_view)
     can :read, Package if /A|Y/.match?(current_user.package_manage)
     can :read, PackageFile if /A|Y/.match?(current_user.package_manage)
+    can :read, Sal3BatchRequestsBc if /A|Y/.match?(current_user.sal3_batch_req)
   end
 
   def assign_staff_manage_permission(current_user)
     can :manage, AccessionNumberUpdate if /A|Y/.match?(current_user.accession_number)
     can :manage, EdiInvoice if /A|Y/.match?(current_user.edi_inv_manage)
     can :manage, EdiErrorReport if /A|Y/.match?(current_user.edi_inv_manage)
```

Comment:
If I have edi_inv_view = "Y" I cannot view the EDI error reports. Is that okay? Or is it expected that I would have edi_inv_manage = "Y" in order to view errors?

Reply:
If you have edi_inv_view = 'Y' you SHOULD be able to view the EDI error reports.  (Otherwise there's not much to do as a "view" user.)

Reply:
I added some changes to allow staff with edi_inv_view to read the error reports.

Reply:
I can now view errors with edi_inv_view='Y'. I can also edit by going directly to `edi_inv_line/837865/edit` even though I have edi_inv_manage=null. I believe I can edit because I have mgt_rpts='A'.

Reply:
I tried it with the `atest` user (with edi_inv_manage: "") and cancan prevents the edit.

Reply:
It works now. Thanks.